incubator/hnc/e2e/tests/delete-anchor-crd_test.go

```diff
@@ -0,0 +1,55 @@
+package test
+
+import (
+	"os/exec"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Delete-anchor-crd", func() {
+	It("should create parent and deletable child, and delete the CRD", func() {
+		// create a namespace delete-crd-parent
+		cmd := exec.Command("kubectl", "create", "ns", "delete-crd-parent")
+		stdout, err := cmd.CombinedOutput()
+		GinkgoT().Log(string(stdout))
+		Expect(err).Should(BeNil())
+
+		// create a namespace delete-crd-child and set its parent as delete-crdparent
+		cmd = exec.Command("kubectl", "hns", "create", "delete-crd-child", "-n", "delete-crd-parent")
+		stdout, err = cmd.CombinedOutput()
+		GinkgoT().Log(string(stdout))
+		Expect(err).Should(BeNil())
+
+		// allow cascading delete for delete-crd-child
+		cmd = exec.Command("kubectl", "hns", "set", "delete-crd-child", "-a")
+		stdout, err = cmd.CombinedOutput()
+		GinkgoT().Log(string(stdout))
+		Expect(err).Should(BeNil())
+
+		// delete the crd
+		cmd = exec.Command("kubectl", "delete", "customresourcedefinition/subnamespaceanchors.hnc.x-k8s.io")
+		stdout, err = cmd.CombinedOutput()
+		GinkgoT().Log(string(stdout))
+		Expect(err).Should(BeNil())
+
+		// verify that the namespace still exists
+		cmd = exec.Command("kubectl", "get", "ns", "delete-crd-child")
+		stdout, err = cmd.CombinedOutput()
+		GinkgoT().Log(string(stdout))
+		Expect(err).Should(BeNil())
+
+		// cleaning up: fix hnc
+		cmd = exec.Command("kubectl", "apply", "-f", "manifests/hnc-manager.yaml")
+		stdout, err = cmd.CombinedOutput()
+		GinkgoT().Log(string(stdout))
+		Expect(err).Should(BeNil())
+
+		// cleaning up: delete namespace
+		// Note: the namespace delete-crd-child could not be deleted due to a known issue #847
```

Comment:
`delete-crd-child` is a [subnamespace](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/docs/user-guide/concepts.md#basic-subns), which means it has the `hnc.x-k8s.io/subnamespaceOf` [annotation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/). You should be able to remove this as follows:

`kubectl annotate namespace delete-crd-child hnc.x-k8s.io/subnamespaceOf-`

Note the "minus" at the end of that line - that means "remove the annotation." Then you should be able to delete the namespace normally.

Reply:
Done

Reply:
Done

Reply:
Couple of things here:

* Probably better to delete the child first, then the parent. In HNC, having a child can sometimes prevent the parent from being deleted.
* You should document _why_ you made the change that we recommended. You added the code, but will the next person to read the test understand why it's there?